### PR TITLE
Automated(ish) cherry pick of #24645

### PR DESCRIPTION
**Pull request by j3ffml**
Not so automatic Cherry pick of #24645 on release-1.2.

### CODE REVIEW

File: test/e2e/util.go [hunk 1]
```diff
@@ -1484,22 +1484,22 @@ func validateController(c *client.Client, containerImage string, replicas int, c
 
 	By(fmt.Sprintf("waiting for all containers in %s pods to come up.", testname)) //testname should be selector
 waitLoop:
-	for start := time.Now(); time.Since(start) < podStartTimeout; time.Sleep(5 * time.Second) {
-		getPodsOutput := runKubectlOrDie("get", "pods", "-o", "template", getPodsTemplate, "--api-version=v1", "-l", testname, fmt.Sprintf("--namespace=%v", ns))
+	for start := time.Now(); time.Since(start) < PodStartTimeout; time.Sleep(5 * time.Second) {
```
Comment (roberthbailey) [inline test/e2e/util.go]: Why did this file change? It wasn't changed in the original PR and these changes seem unrelated to the rest of the cherry pick.

Comment (jlowdermilk) [inline test/e2e/util.go]: It was changed in the [original PR](https://github.com/kubernetes/kubernetes/pull/24645/files). I thought about discarding this change but it seemed wrong to do a partial cherry pick of 24645.

**Comment by janetkuo** [comment]
The original PR #24645 needs the `cherrypick-approved` label, right?

**Comment by roberthbailey** [comment]
@janetkuo Yes, I forgot to add it. Just rectified that.

**Comment by jlowdermilk** [comment]
The test this added (and I suspect the command itself) is flaky. Closing this until fixed #25140.
